<?xml version="1.0"?>
<launch>
    <arg name="robot_name" default="taser"/>
    <arg name="arm_group_name" default="left_arm"/>
    <!-- arm motion service -->
    <group if="$(eval arg('robot_name') == 'taser')">
    <node pkg="arm_motion_service" type="arm_motion_server.py"
          name="arm_motion_service_$(arg robot_name)_$(arg arm_group_name)" required="true" output="screen">
        <param name="arm_group_name" value="$(arg arm_group_name)"/>
        <param name="world_frame" value="taser/base_link"/>
        <param name="max_velocity_scaling_factor" value="1"/>
        <param name="max_acceleration_scaling_factor" value="1"/>
    </node>
    </group>
    <group if="$(eval arg('robot_name') == 'kuka')">
    <node pkg="arm_motion_service" type="arm_motion_server.py"
          name="arm_motion_service_$(arg robot_name)_arm" required="true" output="screen">
        <param name="arm_group_name" value="arm"/>
        <param name="world_frame" value="world"/>
        <param name="max_velocity_scaling_factor" value="0.2"/>
        <param name="max_acceleration_scaling_factor" value="0.2"/>
    </node>
    </group>
    <group if="$(eval arg('robot_name') == 'ur10e')">
    <node pkg="arm_motion_service" type="arm_motion_server.py"
          name="arm_motion_service_$(arg robot_name)_$(arg arm_group_name)" required="true" output="screen">
        <param name="arm_group_name" value="$(arg arm_group_name)"/>
        <param name="world_frame" value="world"/>
        <param name="max_velocity_scaling_factor" value="0.1"/>
        <param name="max_acceleration_scaling_factor" value="0.1"/>
    </node>
    </group>
</launch>
